content/docs/guides/revocation.md

```diff
@@ -65,6 +74,149 @@ check if time($date), $date <= 2018-12-20T00:00:00+00:00;
+fail too. A stateless approach will decouple session creation from authorization, so existing
+sessions can still work when the authentication service fails.
+- **isolation**: the service receiving the request might be less trusted and should not be
+able to access session information.
+- **authentication delegation**: authentication could be in a separate service (example: SSO)
+that can't be queried on every request. That service could even be managed by a different
+company.
+
+In those cases, separating authorization from session creation makes sense, but then how do
+we close a session? It is usually done through token revocation: the authorizer needs to
+know a list of tokens that must be refused, and that list changes dynamically, so we are
+reintroducing some state in the system.
+
+But revocation has properties that make it nice to implement in stateless architectures:
+- we do not need to know about all of the tokens, only those that were revoked, so the list
+will be much smaller
+- the list of revoked tokens will naturally grow, but if tokens have an expiration date, they
+can be purged from the list after a while
+- it is read-oriented and highly cacheable: once a token was added to the revocation list,
+we won't modify its entry (except when purging), so we don't need synchronization or consensus
+- revocation lists do not hold any critical or private information, they can be shared with
+every service
+
+So handling revocation is adding some shared state, but much more limited than what we would
+have with a fully centralized architecture.
+
+## How to implement revocation in our infrastructure?
+
+We need a reliable way to transmit revocation information to services. That will depend on
+how quickly we want to disseminate it, and how much complexity we can bear.
+
+### The basic solution: read the revocation list at startup
+
+In some cases, like communication between automated services, revocation is rare, mostly when
+a service is stopped or a token is leaked, so the revocation list is mostly static and small. If we can accept some
+manual operations, and a (slight) delay in synchronization, we can have services read the revocation list at startup. They
+will check tokens from an in memory list, that will stay the same for the entire life of the
+service (until it is restarted).
+
+The tradeoff here is that if we need to revoke a token urgently, we will need to redeploy
+a lot of services at once.
+
+In the case where there is only one service accepting tokens, the revocation list can be read from config (a config file or environment variables).
+
+In the case where more services accept tokens, it will become necessary to have a centrally
+defined list that is then distributed to all services.
+Since the revocation list is small and changes rarely, it can be stored as a file in an object
+store like S3, and downloaded via HTTP. That file can be updated independently whenever
```

Comment:
This is what Terraform do when dealing with sharing state. But it adds a KV store for locking resource for concurrent access

Reply:
the nice thing here is that it does not need much locking, there won't be concurrent writes